File: 5. Longest Palindromic Substring/5. Longest Palindromic Substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def expandAroundCenter(left, right):
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
            return s[left+1:right]
    
        longest = ""

        for i in range(len(s)):
            # Odd length palindrome
            temp = expandAroundCenter(i, i)
            if len(temp) > len(longest):
                longest = temp
            
            # Even length palindrome
            temp = expandAroundCenter(i, i+1)
            if len(temp) > len(longest):
                longest = temp
            
        return longest

File: 5. Longest Palindromic Substring/5. Longest Palindromic Substring.js
// /**
//  * @param {string} s
//  * @return {string}
//  */

var longestPalindrome = function(s) {
    const expandAroundCenter = (left, right) => {
        while (left >= 0 && right < s.length && s[left] == s[right]){
            left = left - 1
            right = right + 1
        }
        return s.slice(left + 1, right)
    }

    let longest = ""

    for (let i = 0; i < s.length; i++){
        // Odd length palindrome
        let oddTemp = expandAroundCenter(i, i)
        if (oddTemp.length > longest.length) longest = oddTemp

        // Even length palindrome
        let evenTemp = expandAroundCenter(i, i+1)
        if (evenTemp.length > longest.length) longest = evenTemp

    }

    return longest

};